test/controllers/agent/WhatToDoControllerSpec.scala

```diff
@@ -124,32 +125,36 @@ class WhatToDoControllerSpec extends ControllerBaseSpec with MockCustomerDetails
 
         mockAgentAuthorised()
 
-        val result: Future[Result] = controller.submit("l'biz", nonMTDfB = true)(fakeRequestWithVrnAndRedirectUrl
+        val result: Future[Result] = controller.submit(fakeRequestWithVrnAndRedirectUrl
           .withFormUrlEncodedBody("option" -> "view-certificate")
         )
 
         redirectLocation(result) shouldBe Some(mockConfig.vatCertificateUrl)
       }
     }
-    "redirect to the capture preferences page" when {
-      "the email preference is not present in the session" in new Test {
+    "make a call to the customerDetailsService and render the page" when {
```

Comment:
This test doesn't do what the title is saying - its submitting valid information and redirecting to return deadlines